ts/torch_handler/base_handler.py

```diff
@@ -77,14 +77,40 @@ def initialize(self, ctx):
 
         self.initialized = True
 
-    @abc.abstractmethod
     def preprocess(self, data):
-        pass
+        """
+        Override to customize the pre-processing
+        :param data: input data
+        :return: input data
+        """
+        return data
 
-    @abc.abstractmethod
     def inference(self, data):
-        pass
+        """
```

Comment:
The interface (a) is still unclear and (b) requires an override almost always of pre- and post-process. Instead, provide a default implementation that clarifies what a developer should do in each step AND could work without any of the other steps being overridden.

How about this interface?
1. `preprocess`: accepts `list`, returns `Tensor`
2. `inference`: accepts `Tensor`, returns `Tensor`
3. `postprocess`: accepts `Tensor`, returns `list`

See https://github.com/htappen/serve/blob/master/ts/torch_handler/base_handler.py#L144

Reply:
1. preprocess - agreed! 

2. inference - Can't we just replace torch.as_tensor(data) with torch.as_tensor(data, device=self.device) in preprocess itself? Thus we will be left with 
```
results = self.model(torch_data)
marshalled_results = results.to('cpu')
```

3. posprocess - BaseHandler is generic handler and we should assume that it will be only/related to classification hence topk is not required at basehandler.  In fact, it is part of image_classification in this PR and user has ability to override that class/topk.  Regarding `mapping`, this can be model specific and may/may not required mapping. Hence it should be part of specific handlers. Thats why we were just loading it as json file.

Reply:
RE (2): I'd suggest we don't require users to marshal to a certain device. The device initialization code is, as it should be, handled automatically in "initialize". Most users, then, might not be aware of the need to put the data to a certain device. So, let's keep the step in (2) of copying the tensor to the GPU. Note: if the user _does_ already move it to the device in preprocess, then as_tensor(...) becomes a no-op (PyTorch handles that)

RE (3): I can get behind making topk and mapping not included in postprocess, but they do need to be refactored into helper methods as part of BaseHandler, given how many of the example classes use them.

Reply:
@htappen Can you make these changes and create PR against this branch (issue_411). Then I will go ahead and merge your changes to this branch.